internal/jobs/update-recommendations.go

```diff
@@ -0,0 +1,145 @@
+package jobs
+
+/*
+This job looks for missing recommendations, requests and saves them
+Originally intended for integration with Kruize v20
+*/
+
+import (
+	"fmt"
+	"encoding/json"
+	"net/http"
+	"time"
+	"reflect"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils"
+	"github.com/redhatinsights/ros-ocp-backend/internal/model"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils/kruize"
+	
+	database "github.com/redhatinsights/ros-ocp-backend/internal/db"
+)
+
+var log = logging.GetLogger()
+var cfg *config.Config = config.GetConfig()
+var db = database.GetDB()
+
+var (
+	invalidRecommendationUpdateRecommendationJob = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "rosocp_invalid_recommendation_update_recommendation_job",
+		Help: "Invalid recommendations sum from Kruize gathered while executing update-recommendations job",
+	})
+)
+
+func checkURLStatus() bool {
+
+	url := cfg.KruizeUrl + "/updateRecommendations"
+    resp, err := http.Get(url)
+    if err != nil {
+        return false
+    }
+    defer resp.Body.Close()
+
+    return resp.StatusCode == http.StatusOK
+}
+
+func UpdateRecommendations(){
+
+	maxRetries := 10
+    retryInterval := 2 * time.Second
+
+    for i := 0; i < maxRetries; i++ {
+        if checkURLStatus() {
+            RenewRecommendations()
+            break
+        } else {
+            fmt.Printf("/updateRecommendations is down: (Retry %d of %d)\n", i+1, maxRetries)
+            time.Sleep(retryInterval)
+        }
+    }
+
+}
+
+func RenewRecommendations() {
+
+	var workloads []model.Workload
+	query := fmt.Sprintf("SELECT id, experiment_name, metrics_uploaded_at FROM %s", "workloads")
+    if err := db.Raw(query).Scan(&workloads).Error; err != nil {
+        log.Fatal(err)
+    }
+
+    // Process the results
+    for _, workload := range workloads {
+
+		recommendationSets, err := model.GetFirstRecommendationSetsByWorkloadID(workload.ID)
+		
+		if err != nil {
+			panic(err.Error())
+		}
+
+		if !reflect.ValueOf(recommendationSets).IsZero() {
```

Comment:
Can you explain why there is a need to this check over here? and anyway we will be running this job after truncating the recommendationSets table that means this will every time evaluate to `False` and `else` block will be executed.

Reply:
- The exclamation symbol is more of a typo, I will revert that change.
- This check has been added to make the job idempotent, so that we can use this job in future if required.
- The recommendationSets table will be empty yes, but this job will look for recommendations for existing workloads, and then update those specifically

Reply:
Change pushed